### PR TITLE
Rate limiting for requesting signatures and deleting old claimed moves.

### PR DESCRIPTION
**Pull request by alex-lindenbaum**

### CODE REVIEW

File: contracts/src/NStates.sol [hunk 1]
```diff
@@ -136,6 +140,10 @@ contract NStates {
                 tileCommitments[moveInputs.hTTo],
             "Old tile states must be valid"
         );
+        // currentBlockHeight > c.blockSubmitted + CLAIMED_MOVE_LIFE_SPAN
```
Comment (lyronctk) [inline contracts/src/NStates.sol:143]: remove pls :)

Comment (lyronctk) [reply contracts/src/NStates.sol:143]: wait actually, why is this commented out lol

File: enclave/server.ts [hunk 2]
```diff
@@ -123,23 +144,33 @@ async function login(
...
+            const digest = utils.solidityKeccak256(
+                ["uint256", "uint256", "uint256"],
+                [currentBlockHeight, hUFrom, hUTo]
+            );
+            const sig = await signer.signMessage(utils.arrayify(digest));
 
-    socket.emit("getSignatureResponse", sig, uFrom, uTo);
+            socket.emit("signatureResponse", sig, currentBlockHeight);
+        } else {
+            socket.emit("errorResponse", "Cannot move more than once per tick");
```
Comment (lyronctk) [inline enclave/server.ts:171]: should cut the connection- this is a malicious client 

there should also be client-side code for limiting to one move per block

File: contracts/src/NStates.sol (hunk 1 above)
Comment (alex-lindenbaum) [reply contracts/src/NStates.sol:143]: forgot to commit oops

File: enclave/server.ts (hunk 2 above)
Comment (alex-lindenbaum) [reply enclave/server.ts:171]: I think if a mod accidentally submits +1 move in a block, they shouldn't be booted immediately, no?

Comment (alex-lindenbaum) [reply enclave/server.ts:171]: or do we be very strict with client-side code for limiting moves / block

Comment (lyronctk) [reply enclave/server.ts:175]: yes strict & have client side code 

this pattern's common- having an exact definition of honest client behavior and terminating at any deviation

**Review by lyronctk — APPROVED** [review]

**Pull request merged**